linera-witty/src/type_traits/implementations/std/mod.rs

```diff
@@ -3,14 +3,101 @@
 
 //! Implementations of the custom traits for types from the standard library.
 
+/// A macro to implement [`WitType`][super::WitType], [`WitLoad`][super::WitLoad] and
+/// [`WitStore`][super::WitStore] for a simple wrapper type.
+///
+/// This assumes that:
+/// - The type is `$wrapper<T>`
+/// - It can be constructed with `$wrapper::new(instance_to_wrap)`
+/// - The type implements `Deref<Target = T>`
+macro_rules! impl_for_wrapper_type {
+    ($wrapper:ident) => {
+        impl<T> crate::WitType for $wrapper<T>
+        where
+            T: crate::WitType,
+        {
+            const SIZE: u32 = T::SIZE;
+
+            type Layout = T::Layout;
+            type Dependencies = T::Dependencies;
+
+            fn wit_type_name() -> ::std::borrow::Cow<'static, str> {
+                T::wit_type_name()
```

Comment:
Why not show the wrapper as well (`$wrapper<T::wit_type_name()>`)?

Reply:
Because the wrapper types aren't part of the [WIT standard](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md#wit-types) :/

Reply:
Hmm, but that leads me to a different thought. This should be empty, and the `T` should be a dependency :thinking: I'll fix it, thanks!

Reply:
Nope, I was wrong. The problem is that if I do that, then the same `wit_type_name` is used for two different types (`T` and `Box<T>`), and they end up having different `wit_type_declaration`s, which is a problem when generating the `.wit` files. So I had to revert to make `Box<T>` identical to `T` for it to work.